### PR TITLE
Add QAT support for 420xx driver and the 4946 device (GNR-D)

### PR DESCRIPTION
**Pull request by tkatila**
Fixes: #1644

### CODE REVIEW

File: cmd/qat_plugin/dpdkdrv/dpdkdrv.go [hunk 1]
```diff
@@ -381,8 +382,10 @@ func readDeviceConfiguration(pfDev string) string {
 		IgnoreInlineComment: true,
 	}
 
+	driver := getCurrentDriver(pfDev)
+
 	devCfgPath := filepath.Join(filepath.Dir(filepath.Join(pfDev, "../../")), "kernel/debug",
-		fmt.Sprintf("qat_4xxx_%s/dev_cfg", filepath.Base(pfDev)))
+		fmt.Sprintf("qat_%s_%s/dev_cfg", driver, filepath.Base(pfDev)))
```
Comment (mythi) [inline cmd/qat_plugin/dpdkdrv/dpdkdrv.go:388]:
```suggestion
		fmt.Sprintf("qat_%s_%s/dev_cfg", getCurrentDriver(pfDev), filepath.Base(pfDev)))
```

**Comment by mythi** [comment]
QAT README and 'checking for hardware' could be made up-to-date with this as well

File: .github/workflows/lib-e2e.yaml [hunk 2]
```diff
@@ -27,6 +27,13 @@ jobs:
               - intel-idxd-config-initcontainer
               - accel-config-demo
               - intel-deviceplugin-operator
+          - name: e2e-qat-gnrd
+            targetjob: e2e-qat FOCUS="Mode:dpdk.*Resource:(cy|dc)" SKIP=App:crypto-perf
+            runner: simics-gnrd
+            images:
+              - intel-qat-plugin
+              - intel-qat-initcontainer
+              - openssl-qat-engine
```
Comment (hj-johannes-lee) [inline .github/workflows/lib-e2e.yaml:36]: If we try running crypto-perf that I mentioned above..
```suggestion
              - openssl-qat-engine
              - crypto-perf
```

File: .github/workflows/lib-e2e.yaml [hunk 3]
```diff
@@ -27,6 +27,13 @@ jobs:
               - intel-idxd-config-initcontainer
               - accel-config-demo
               - intel-deviceplugin-operator
+          - name: e2e-qat-gnrd
+            targetjob: e2e-qat FOCUS="Mode:dpdk.*Resource:(cy|dc)" SKIP=App:crypto-perf
```
[the remark this answers is not in the document]
Comment (tkatila) [reply .github/workflows/lib-e2e.yaml:31]: Early signs seem to indicate that at least the current crypto-perf image would not run on. But I don't yet have a definitive answer.

[the remark this answers is not in the document]
Comment (hj-johannes-lee) [reply .github/workflows/lib-e2e.yaml:31]: I see. Then, I am ok without that. Maybe later I can try to add!

File: cmd/qat_plugin/README.md [hunk 4]
```diff
@@ -132,7 +132,7 @@ In addition to the default configuration, you can add device-specific configurat
 
 | Device | Possible Configuration | How To Customize | Options | Notes |
 |:-------|:-----------------------|:-----------------|:--------|:------|
-| 4xxx, 401xx,402xx | [cfg_services](https://github.com/torvalds/linux/blob/v6.6-rc5/Documentation/ABI/testing/sysfs-driver-qat) reports the configured services (crypto services or compression services) of the QAT device. | `ServicesEnabled=<value>` | compress:`dc`, crypto:`sym;asym`, <br>crypto+compress:`asym;dc`,<br>crypto+compress:`sym;dc` | Linux 6.0+ kernel is required. |
+| 4xxx, 401xx, 402xx, 420xx | [cfg_services](https://github.com/torvalds/linux/blob/v6.6-rc5/Documentation/ABI/testing/sysfs-driver-qat) reports the configured services (crypto services or compression services) of the QAT device. | `ServicesEnabled=<value>` | compress:`dc`, crypto:`sym;asym`, <br>crypto+compress:`asym;dc`,<br>crypto+compress:`sym;dc` | Linux 6.0+ kernel is required. |
```
Comment (hj-johannes-lee) [inline cmd/qat_plugin/README.md:135]: Do you think it would be good to write that for 420xx even 6.8+ is needed?

Comment (tkatila) [reply cmd/qat_plugin/README.md:135]: Good point. It would be good to note that.

**Review by mythi — APPROVED** [review]

**Review by hj-johannes-lee — APPROVED** [review]

**Pull request merged**